### ISSUE TITLE
Slow REL file operations on real hardware

### ISSUE THREAD

**Issue by fachat** [fachat/XD2031#179]
On real hardware REL file operations so slow that the IEEE488 timeout needs to be disabled to actually make them work.
Observed by @snhirsch

**Comment by fachat** [comment]
I have a setup with a PET, an early petSD2, and an SFD1001. I tried some tests creating and reading REL files, but never needed to set the timeout to make the REL file operations work. 

Do you have a logic analyzed? Can you trace the IEEE488 bus (200kHz sample freq usually works) while such a timeout happens and send me the files (separately)?

**Comment by fachat** [comment]
Did you actually set the fuses right for XD2031 on the petSD? In the firmware/petSD there is a README that hints on data loss on the IEC bus ... not sure if it may affect IEEE488, I never had a petSD.
I will have a further look at the hardware differences and what they mean for the firmware.

**Comment by snhirsch** [comment]
I had sent you such a trace last fall :-).  Let me do some digging and see if I can find the data.

**Comment by fachat** [comment]
sorry that I forgot about that. Anyway, did you check the fuses?

**Comment by snhirsch** [comment]
That's a sore subject.  I bricked one MPU already by fiddling with fuses and would rather not repeat this exercise. What exact setting(s) are you referring to?

**Comment by fachat** [comment]
I am talking about this README https://github.com/fachat/XD2031/blob/master/firmware/petSD/README - must be by @nils-eilers  as I've never had a petSD.

**Comment by snhirsch** [comment]
That would have been done, yes.

**Comment by fachat** [comment]
just to be sure - you are using this device? http://petsd.net/orgpetsd.php?lang=en

**Comment by snhirsch** [comment]
Yes, that's it.  The only difference is that mine lacks clock and ethernet.

**Comment by fachat** [comment]
hm. I have an intermediate version "petSD2" between that and petSD+ that lacks Ethernet but has a clock (at least it has a coin cell)....... can you pls verify that pin 1 of the 75160 or 75161 (TE signal) is connected to pin 1 (PB0) of the Atmega 1284? (or even check all signals as defined in firmware/petSD/hwdefines.h?

**Comment by fachat** [comment]
Also, does the problem also appear with only the petSD attached to the PET (and the other drive(s) completely disconnected, not just switched off)?

**Comment by fachat** [comment]
There are two operations / test cases that usually test the IEEE488 communication pretty well: using the DIRECTORY (or CATALOG) command on BASIC4, as this basically does a TALK/SECTALK/GET/UNTALK for every byte it reads - and a small test program I have written that dumps some data with single-character PRINT#. As DIR seems to work (or you'd probably have reported it already), can you also check and run this program https://github.com/fachat/XD2031/blob/180-rel-len/tests/seq.lst an see if the result is an 8k file that has just counting bytes like 00 01 02 ... fe ff 00 01 ... in it, and no byte missing?

**Comment by snhirsch** [comment]
That test passes without incident.  All 256-byte chunks are identical.
